client/my-sites/people/people-section-nav/index.jsx

```diff
@@ -51,63 +59,66 @@ let PeopleNavTabs = React.createClass( {
 			</NavTabs>
 		);
 	}
-} );
+}
 
-module.exports = React.createClass( {
+class PeopleSectionNav extends React.PureComponent {
+	constructor( props ) {
+		super( props );
+	}
 
-	displayName: 'PeopleSectionNav',
+	canSearch() {
+		const { site, filter } = this.props;
 
-	canSearch: function() {
 		// Disable search for wpcom followers and viewers
-		if ( this.props.filter ) {
-			if ( 'followers' === this.props.filter || 'viewers' === this.props.filter ) {
+		if ( filter ) {
+			if ( 'followers' === filter || 'viewers' === filter ) {
 				return false;
 			}
 		}
 
-		if ( ! this.props.site.jetpack ) {
+		if ( ! site.jetpack ) {
 			// wpcom sites will always support search
 			return true;
 		}
-
-		if ( 'team' === this.props.filter && ! this.props.site.versionCompare( '3.7.0-beta', '>=' ) ) {
+		if ( 'team' === filter && versionCompare( site.options.jetpack_version, '3.7.0-beta' ) < 0 ) {
 			// Jetpack sites can only search team on versions of 3.7.0-beta or later
 			return false;
 		}
 
 		return true;
-	},
+	}
 
-	getFilters: function() {
-		var siteFilter = this.props.site.slug,
+	getFilters() {
+		const { site } = this.props,
+			siteFilter = ( site && site.slug ) ? site.slug : '',
 			filters = [
 				{
-					title: this.translate( 'Team', { context: 'Filter label for people list' } ),
+					title: this.props.translate( 'Team', { context: 'Filter label for people list' } ),
 					path: '/people/team/' + siteFilter,
 					id: 'team'
 				},
 				{
-					title: this.translate( 'Followers', { context: 'Filter label for people list' } ),
+					title: this.props.translate( 'Followers', { context: 'Filter label for people list' } ),
 					path: '/people/followers/' + siteFilter,
 					id: 'followers'
 				},
 				{
-					title: this.translate( 'Email Followers', { context: 'Filter label for people list' } ),
+					title: this.props.translate( 'Email Followers', { context: 'Filter label for people list' } ),
 					path: '/people/email-followers/' + siteFilter,
 					id: 'email-followers'
 				},
 				{
-					title: this.translate( 'Viewers', { context: 'Filter label for people list' } ),
+					title: this.props.translate( 'Viewers', { context: 'Filter label for people list' } ),
 					path: '/people/viewers/' + siteFilter,
 					id: 'viewers'
 				}
 			];
 
 		return filters;
-	},
+	}
 
-	getNavigableFilters: function() {
-		var allowedFilterIds = [ 'team' ];
+	getNavigableFilters() {
+		let allowedFilterIds = [ 'team' ]; // eslint-disable-line prefer-const
```

Comment:
Why are we not using const here? As far as I'm reading this, the ref doesn't get updated again in this function. Note that in js const only guards against ref changes, and doesn't grant immutability